enhancements/testing/improved-platform-tests.md

```diff
@@ -0,0 +1,438 @@
+extracting the test binaries, and using those binaries for running tests.
+The followup work requires:
+
+1. Extracting common test library for re-use. The library should be based on the
+   contents of [k8s-tests directory](https://github.com/openshift/kubernetes/tree/8bd5514be14de7fb3105962c8fc206ef5e921a0b/openshift-hack/cmd/k8s-tests).
+2. Expanding [externalTestsForSuite](https://github.com/openshift/origin/blob/cc42164781a728b804d0bb07f09cf878ec8f7807/pkg/test/ginkgo/external.go#L29)
+   to allow extracting test binaries from other images, currently it uses hardcoded
+   `hyperkube` image from the release payload, but should rather rely on [an annotation](#implementation-detailsnotesconstraints),
+   telling which images contain the external test binary, and its location inside
+   the image.
+3. Preparing a template/demo for one of operators building on top of steps 1 and 2
+   above.
+
+### Even more future work
+
+As mentioned in [risks section](#risks-and-mitigations), upgrade tests are relying
+on the vendored code only, it would be reasonable to address this issue in the long,
+to run the upgrade tests in a similar fashion as the other tests.
+
+Re-evaluate [goal #4](#goals) which explicitly requires providing a fallback for
+kubernetes vendored tests. Currently, at least Microshift requires us to have
+the kubernetes tests vendored since they don't provide release image from which
+we can extract the kubernetes test binaries.
+
+### Test Plan
+
+Not applicable, since this is changing testing mechanism. Prior landing this PR
+we will ensure that all payload tests are passing.
+
+### Graduation Criteria
+
+N/A
+
+#### Dev Preview -> Tech Preview
+
+N/A
+
+#### Tech Preview -> GA
+
+N/A
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+N/A
+
+### Version Skew Strategy
+
+N/A
+
+### Operational Aspects of API Extensions
+
+N/A
+
+#### Failure Modes
+
+N/A
+
+#### Support Procedures
+
+N/A
+
+## Implementation History
+
+- *2022-11-16* - Initial draft.
+- *2023-05-31* - Updated document based on the proof of concept implementation.
+
+## Alternatives
+
+### Runtime Assembly - Multiple pods
+
+Test binaries are built into each release payload component image at conventionally
+established location; e.g. `/openshift-tests/<component-name>`. Once a cluster
+has been installed, the `openshfit-tests` binary inspects the components of the
+running cluster's payload and creates pods, running those images, with
+`/openshift-tests/<component-name>` binaries as the entrypoint.
+`openshift-tests` would still be responsible for aggregating the results.
+
+Pros:
+- It’s “easy” to have a PR add a new test and that test will run in pre-merge testing.
+- It’s easier to implement, since the coordination mechanism will exist only inside origin repository.
+
+Cons:
+- An amd64 build farm cluster cannot run release payload component pods from an ARM cluster.
+
+### Build-time Assembly - Aggregating Source
```

Comment:
there's another build-time assembly alternative:  aggregating distinct binaries into a single test image

it's the option i prefer and i'd like to see more details around why we wouldn't/can't do it that way.

Reply:
Added.